buildflow/core/node/node.py

```diff
@@ -84,25 +92,59 @@ def wrapper_function(*args, **kwargs):
 class Node(NodeAPI):
     def __init__(
         self,
-        name: str = "",
+        node_id: NodeID = "",
         runtime_config: RuntimeConfig = RuntimeConfig.DEBUG(),
         infra_config: InfraConfig = InfraConfig.DEBUG(),
     ) -> None:
-        self.name = name
+        self.node_id = node_id
         self._processors: List[Processor] = []
         # The Node class is a wrapper around the Runtime and Infrastructure
         self._runtime_config = runtime_config
-        self._runtime = None
-        self._infra = PulumiInfraActor.remote(infra_config)
+        self._runtime_actor = None
+        self._infra_config = infra_config
 
-    def processor(self, source: SourceType, sink: Optional[SinkType] = None, **kwargs):
+    def processor(
+        self,
+        source: SourceType,
+        sink: Optional[SinkType] = None,
+        num_cpus: float = 1.0,
+        num_concurrent_tasks: int = 1,
+        log_level: str = None,
+    ):
         # NOTE: processor_decorator is a function that returns an Ad Hoc
         # Processor implementation.
-        return processor_decorator(node=self, source=source, sink=sink, **kwargs)
+        return _processor_decorator(
+            node=self,
+            source=source,
+            sink=sink,
+            num_cpus=num_cpus,
+            num_concurrent_tasks=num_concurrent_tasks,
+            log_level=log_level,
+        )
 
-    def add(self, processor: Processor):
-        if self._runtime is not None:
+    def add(
+        self,
+        processor: Processor,
+        *,
+        num_cpus: float = 1.0,
+        num_concurrent_tasks: int = 1,
+        log_level: str = None,
+    ):
+        if self._runtime_actor is not None and self._runtime_actor.is_active():
             raise RuntimeError("Cannot add processor to running node.")
+        if processor.processor_id in self._runtime_config.replica_configs:
+            raise RuntimeError(
+                f"Processor({processor.processor_id}) already exists in node."
+            )
+        if log_level is None:
+            log_level = self._runtime_config.log_level
+
+        # Each processor gets its own replica config
+        self._runtime_config.replica_configs[processor.processor_id] = ReplicaConfig(
+            num_cpus=num_cpus,
+            num_concurrent_tasks=num_concurrent_tasks,
+            log_level=self._runtime_config.log_level,
```

Comment:
nit: I think this should just be `log_level` looks like it'll get set to this above

Reply:
done